dev/com.ibm.ws.org.apache.cxf.cxf.rt.databinding.jaxb.3.2/src/org/apache/cxf/jaxb/JAXBDataBinding.java

```diff
@@ -457,13 +520,28 @@ protected void justCheckForJAXBAnnotations(ServiceInfo serviceInfo) {
         }
     }
     private void checkForJAXBAnnotations(MessagePartInfo mpi, SchemaCollection schemaCollection, String ns) {
+
+        if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+           LOG.fine("checkForJAXBAnnotations for MessagePart:" + (mpi != null ? mpi.getName() : "null") );
+        } // Liberty change end
+
         Annotation[] anns = (Annotation[])mpi.getProperty("parameter.annotations");
+        if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+	   if (anns != null && anns.length > 0) {
+	      for (Annotation an1 : anns) {
+                 LOG.fine("checkForJAXBAnnotations: Found annotation: " + an1);
```

Comment:
You might also get the contents of the annotations when logging just the mpi, which would save you having to for loop here.

Reply:
yes, good point, will replace for loop with just logging annotations array